crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -86,6 +86,11 @@ impl<'interner> TypeChecker<'interner> {
                         let len = Type::Constant(string.len() as u64);
                         Type::String(Box::new(len))
                     }
+                    HirLiteral::FmtStr(string, idents) => {
+                        let len = Type::Constant(string.len() as u64);
```

Comment:
Is this length meaningful? Since it includes the length of the formatting characters `{` and `}` but not of the final formatted string. As a result, is there a problem if we try to do `if a { f"{foo}" } else { f"{bar}" }` if the format strings expand to strings of different lengths?

Reply:
I do need this length to codegen the entire string literal of the format string (https://github.com/noir-lang/noir/blob/a7fa482ed979e890b0e9e29370535f9a401a4ed2/crates/noirc_evaluator/src/ssa_refactor/ssa_gen/context.rs#L185). This string literal is then converted back from fields elements with the `{` and `}` when processing the foreign calls and the format groups are replaced (https://github.com/noir-lang/noir/blob/a7fa482ed979e890b0e9e29370535f9a401a4ed2/crates/nargo/src/ops/foreign_calls.rs#L104). 

However, there will be a problem if you try to do something like `if a { f"{foo}" } else { f"{x}" }`, as `{foo}` and `{x}` are different sizes (5 and 3 respectively). Something like 
```
    let struct_string = if x != 5 { f"{foo}" } else { f"{bar}" };
    std::println(struct_string);
```
will work as expected though